cli.ts

```diff
@@ -41,12 +46,53 @@ sade('lighthouse-parade <url> [dataDirectory]', true)
       // eslint-disable-next-line no-new
       new URL(url);
       const ignoreRobotsTxt: boolean = opts['ignore-robots'];
+      const reportsDirPath = path.join(dataDirectory, 'reports');
+      fs.mkdirSync(reportsDirPath, { recursive: true });
+
       const userAgent: unknown = opts['crawler-user-agent'];
       if (userAgent !== undefined && typeof userAgent !== 'string') {
         throw new Error('--crawler-user-agent flag must be a string');
       }
 
-      scan(url, { ignoreRobotsTxt, dataDirectory, userAgent });
+      const scanner = scan(url, {
+        ignoreRobotsTxt,
+        dataDirectory,
+        shouldRunLighthouseOnURL(url) {
+          if (
+            !fileDoesntExist(makeFileNameFromUrl(url, 'csv'), reportsDirPath)
+          ) {
+            console.log('Skipping report because file already exists');
+            return false;
+          }
+
+          return true;
+        },
+      });
+
+      const urlsFile = path.join(dataDirectory, 'urls.csv');
+      fs.writeFileSync(urlsFile, 'URL,content_type,bytes,response\n');
+      const urlsStream = fs.createWriteStream(urlsFile, { flags: 'a' });
+
+      scanner.on('urlFound', (url, contentType, bytes, statusCode) => {
+        const csvLine = [
+          JSON.stringify(url),
+          contentType,
+          bytes,
+          statusCode,
+        ].join(',');
+        urlsStream.write(`${csvLine}\n`);
+      });
+      scanner.on('reportComplete', (url, reportData) => {
+        console.log('Report is done for', url);
```

Comment:
Recently I've felt like this `console.log` AND the following are redundant... unless we introduced a `--verbose` or `--debug` option or something.

Reply:
I think I'll just keep this first `console.log` if that is fine with you